mitodl/ol-keycloak
# 91: MVP Keycloak UI

# What are the relevant tickets?
https://github.com/mitodl/mit-open/issues/91

# Description (What does it do?)
Updates the ol Keycloak theme to align with the mockups here: https://www.figma.com/proto/LPnWeOp4ALDqhENz15bZZV/Single-Sign-On---SSO?type=design&node-id=204-650&t=0nvLvdJSyB7WQnrx-0&scaling=scale-down&page-id=0%3A1


# Screenshots (if appropriate):
https://www.figma.com/proto/LPnWeOp4ALDqhENz15bZZV/Single-Sign-On---SSO?type=design&node-id=204-650&t=0nvLvdJSyB7WQnrx-0&scaling=scale-down&page-id=0%3A1


# How can this be tested?
After building and copying the theme and extensions jar files, using the instructions described in the readme of this repository, login to the Keycloak Admin Console.  

1. Update the realm login theme to `ol`.
2. Duplicate the browser flow.  Update the duplicate flow to be:
![Screenshot 2023-12-01 at 11 12 39 AM](https://github.com/mitodl/ol-keycloak/assets/8311573/8a32590c-950e-4347-a5b3-d5a22ac5b516)
3. Bind the duplicate browser flow.
4. Create a dummy SAML identity provider (does not need to link to an actual identity provider).
5. Create a second user who is linked to the identity provider and has a verified email address.
6. Setup Keycloak with an SMTP email service.
7. Log out of the Keycloak Admin Console.
8. Test the process flows documented here: https://docs.google.com/document/d/17tJ-C2EwWoSpJWZKjuhMVgsqGtyPH0IN9KakXvSKU0M/edit?pli=1#heading=h.ratoog6xmca6
9. Verify that UI is similar to the mockups described here https://www.figma.com/proto/LPnWeOp4ALDqhENz15bZZV/Single-Sign-On---SSO?type=design&node-id=204-650&t=0nvLvdJSyB7WQnrx-0&scaling=scale-down&page-id=0%3A1
10. Verify that the screens display well on mobile displays.

# Additional Context

1. Some animations and styling are absent due to the complexity to implement them within Keycloak.
